source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveRenderContext.cs

```diff
@@ -465,13 +472,96 @@ public void ToggleShowCardVisibility(Button uiAction)
             {
                 var targetVisibility = card.Visibility == Visibility.Visible ? Visibility.Collapsed : Visibility.Visible;
                 // need to make sure we collapse all showcards before showing this one
-                foreach(var showCard in peers)
+                foreach (var showCard in peers)
                 {
                     showCard.Visibility = Visibility.Collapsed;
                 }
 
-                card.Visibility = targetVisibility; 
+                card.Visibility = targetVisibility;
             }
         }
+
+        private bool ValidateInputs(AdaptiveSubmitAction submitAction)
+        {
+            bool allInputsValid = true, firstInvalidInputFound = false;
+            Dictionary<string, Func<string>> newInputBindings = new Dictionary<string, Func<string>>();
+
+            // We clear the InputBindings collection to clear all the results
+            InputBindings.Clear();
+
+            List<string> inputsToValidate = RetrieveInputList(submitAction);
+
+            // Iterate through all the elements and validate them
+            foreach (string inputId in inputsToValidate)
+            {
+                AdaptiveInputValue inputValue = InputValues[inputId];
+                bool inputIsValid = inputValue.Validate();
+                allInputsValid = allInputsValid && inputIsValid;
+
+                // If the validation failed, set focus to the first element that failed
+                if (!allInputsValid && !firstInvalidInputFound)
```

Comment:
>firstInvalidInputFound [](start = 40, length = 22)

nit: this is perfectly good. it was less clear to me compared to having inputIsValid in the if statement. ex. if allinputs are not valid and first Invalid Input has not been found vs. if input is not valid and first invalid input has